public/javascripts/EditProduct.js

```diff
@@ -127,31 +131,34 @@ var editProduct = {
         //console.log("Description: " + description);
 
         // Associate customers
-        $("#customer_select option:selected").each(function(){
-            var host ="/associateProductCustomer/"
+        $("#customer_select option:selected").each(function () {
+            var host = "/associateProductCustomer/"
                 + window.args.ProductID + "/"
                 + parseInt($(this).val());
 
-            navigation.hit(host, function(response) {
-                if(response != "Success"){
+            navigation.get(host, function (err, response) {
+                if (err) {
+                    $("#message").text("Error: " + err.responseText);
+                }
+                else if (response != "Success") {
                     $("#message").text("Error: " + response);
                 }
-            })/*.fail(function(err) {
-                $("#message").text("Error: " + err.responseText);
-            })*/;
+            })
         });
 
-        // Submit name and description change
-        navigation.hit("/reSubmit/" + window.args.ProductID +"/" + name + "/" + description, function(response){
-            if( response != "Success" ){
+        //submit name and descricption change:
+        navigation.get(window.apiRoute + "/reSubmit/" + window.args.ProductID + "/" + name + "/" + description, function (err, response) {
+            if (err) {
+                $("#message").text("Error: " + err.responseText);
+            }
+            else if (response != "Success") {
                 $("#message").text("Error: " + response);
             }
-            navigation.go("DisplayInventory.html");
```

Comment:
It looks like the original wants to navigate when response != Success, so we'd want to have it on line 156 also.